collector-bitmovin-player/src/main/java/com/bitmovin/analytics/bitmovin/player/BitmovinSdkAdapter.java

```diff
@@ -94,163 +94,182 @@ public void init() {
+            VideoQuality videoQuality = bitmovinPlayer.getPlaybackVideoData();
+            if (videoQuality != null) {
+                data.setVideoBitrate(videoQuality.getBitrate());
+                data.setVideoPlaybackHeight(videoQuality.getHeight());
+                data.setVideoPlaybackWidth(videoQuality.getWidth());
+                data.setVideoCodec(videoQuality.getCodec());
             }
-        }
 
-        //video quality
-        VideoQuality videoQuality = bitmovinPlayer.getPlaybackVideoData();
-        if (videoQuality != null) {
-            data.setVideoBitrate(videoQuality.getBitrate());
-            data.setVideoPlaybackHeight(videoQuality.getHeight());
-            data.setVideoPlaybackWidth(videoQuality.getWidth());
-            data.setVideoCodec(videoQuality.getCodec());
-        }
+            //audio quality
+            AudioQuality audioQuality = bitmovinPlayer.getPlaybackAudioData();
+            if (audioQuality != null) {
+                data.setAudioBitrate(audioQuality.getBitrate());
+                data.setAudioCodec(audioQuality.getCodec());
+            }
 
-        //audio quality
-        AudioQuality audioQuality = bitmovinPlayer.getPlaybackAudioData();
-        if (audioQuality != null) {
-            data.setAudioBitrate(audioQuality.getBitrate());
-            data.setAudioCodec(audioQuality.getCodec());
-        }
+            //Subtitle info
+            SubtitleTrack subtitle = bitmovinPlayer.getSubtitle();
+            if (subtitle != null && subtitle.getId() != null) {
+                data.setSubtitleLanguage(subtitle.getLanguage() != null ? subtitle.getLanguage() : subtitle.getLabel());
+                data.setSubtitleEnabled(true);
+            }
 
-        //Subtitle info
-        SubtitleTrack subtitle = bitmovinPlayer.getSubtitle();
-        if (subtitle != null && subtitle.getId() != null) {
-            data.setSubtitleLanguage(subtitle.getLanguage() != null ? subtitle.getLanguage() : subtitle.getLabel());
-            data.setSubtitleEnabled(true);
-        }
+            //Audio language
+            AudioTrack audioTrack = bitmovinPlayer.getAudio();
+            if (audioTrack != null && audioTrack.getId() != null) {
+                data.setAudioLanguage(audioTrack.getLanguage());
+            }
 
-        //Audio language
-        AudioTrack audioTrack = bitmovinPlayer.getAudio();
-        if (audioTrack != null && audioTrack.getId() != null) {
-            data.setAudioLanguage(audioTrack.getLanguage());
-        }
+            // DRM Information
+            if (drmInformation != null) {
+                data.setDrmType(drmInformation.getType());
+            }
 
-        // DRM Information
-        if (drmInformation != null) {
-            data.setDrmType(drmInformation.getType());
+            return data;
+        } catch (Exception e) {
+            Log.d(TAG, "On Creating Event Data", e);
         }
-
-        return data;
+        return null;
```

Comment:
For me this move looks dangerous...
In `BitmovinAnalytics` we have no null check and therefore would throw another exception
Did we tracked any error during the creation of event Data?

Reply:
This method gets a lot of data from player, so for me there is a great risk that something will possibly go wrong here. ATM if something goes wrong here it will probably be caught by other try catch blocks, but still I wouldn't like to rely on that some caller method will handle exceptions happening here.
I agree that returning null is not the best option, but maybe it's less harm then completely crashing app 🤷‍♀️ 
I'm open to any ideas here (also to remove this try catch completely)